src/niscope/system_tests/test_system_niscope.py

```diff
@@ -125,3 +125,55 @@ def test_waveform_processing(session):
     session.clear_waveform_processing()
     session.horz_record_length == 4096
     session.horz_sample_rate == 10000000
+
+
+def test_configure_ref_levels(session):
+    session.configure_ref_levels()
+    session.commit()
+    assert session.meas_chan_high_ref_level == 90.0
+
+
+# revisit after #630 closed
+def test_configure_trigger_digital(session):
+    session.configure_trigger_digital('VAL_RTSI_0')
+    session.vertical_range = 5
+    assert 'VAL_RTSI_0' == session.trigger_source
+
+
+def test_configure_trigger_edge(session):
+    assert niscope.TriggerSlope.POSITIVE == session.trigger_slope
+    session.configure_trigger_edge('0', niscope.TriggerCoupling.DC)
+    session.commit()
```

Comment:
Calling commit() is unnecessary in most (all?) of these tests. When you read an attribute, the driver will internally verify the session and compute all the coerced values for attributes. commit() is too heavy handed as it _also_ applies settings to hardware (not that it matters much in simulated devices).

Regardless, any redundant code is best removed.

Reply:
Leaving a test with commit to check call to commit method. Removed redundant commit.